ScoutSuite/providers/gcp/facade/facade.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+
+from googleapiclient import discovery
+from ScoutSuite.providers.gcp.utils import MemoryCache
+
+class GCPFacade:
+    def __init__(self):
+        self._resourcemanager_client = discovery.build('cloudresourcemanager', 'v1', cache_discovery=False, cache=MemoryCache())
+
+    # TODO: Make truly async    
+    async def get_projects(self):
```

Comment:
I think you could generalize this method. There is code duplication between this and [this](https://github.com/nccgroup/ScoutSuite/pull/230/files#diff-ffea74d1a65db135e82d200249fd7e5eR21).

Reply:
Yeah, I had noticed but I wasn't sure how to do it cleanly, I think I found a decent way. Let me know what you think!

Reply:
Yup, that's exactly what I had in mind. You'll have to make this async though (like [this commit](https://github.com/nccgroup/ScoutSuite/pull/224/commits/25967e57b29d9c66af0daf339162aa5e0492e3ea)). Michael spent a few days looking for the performance bottleneck, and I spent a day, and it was that; it makes a huge difference.

Reply:
I'm aware, that's what the comment was for, I was waiting for Michael's PR to be merged before adding it! 😉